src/components/AvatarGroup/AvatarGroup.jsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import Avatar from "../Avatar/Avatar";
+import Counter from "../Counter/Counter";
+import Tooltip from "../Tooltip/Tooltip";
+import AvatarGroupCounter from "./AvatarGroupCounter";
+import styles from "./AvatarGroup.module.scss";
+
+const AvatarGroup = ({
+  className,
+  avatarClassName,
+  id,
+  children,
+  size,
+  type,
+  max,
+  counterProps,
+  counterTooltipCustomProps
+}) => {
+  if (!children) {
+    return null;
+  }
+
+  children = React.Children.toArray(children);
+  const displayAvatars = children.slice(0, max);
```

Comment:
this for example need to be in use memo because you want to iterate all the children for slicing only if there is a change on the children

Reply:
Should displayAvatars and counterTooltipAvatars also be memoized?